flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamMultipleInputProcessor.java

```diff
@@ -268,7 +252,12 @@ public InputProcessor(
 		}
 
 		public InputStatus processInput() throws Exception {
-			return taskInput.emitNext(dataOutput);
+			InputStatus status = taskInput.emitNext(dataOutput);
+			if (status == InputStatus.END_OF_INPUT) {
+				dataOutput.endOutput();
```

Comment:
It looks strange: you are asking `taskInput` to emit records to `dataOutput`. It tells you `END_OF_INPUT` and you are calling `endOutput()` on the `dataOutput` that you have just passed to the `taskInput` .

Reply:
After taking the previous comment into consideration, I think this will make sense. The `taskInput` will not be aware of the `endOutput` method any more, the `Processor` will though.